layouts/Base.tsx

```diff
@@ -1,15 +1,73 @@
 'use client';
 
+import classNames from 'classnames';
+import { memo, useMemo, useState } from 'react';
 import type { FC, PropsWithChildren } from 'react';
 
+import WithNavBar from '@/components/withNavBar';
+import WithProgressionSidebar from '@/components/withProgressionSidebar';
+import WithSidebar from '@/components/withSidebar';
+import { usePathname } from '@/navigation.mjs';
 import { NotificationProvider } from '@/providers/notificationProvider';
+import { ShowSidebarProvider } from '@/providers/showSidebarProvider';
 
 import styles from './layouts.module.css';
 
-const BaseLayout: FC<PropsWithChildren> = ({ children }) => (
-  <NotificationProvider viewportClassName="absolute bottom-0 right-0 list-none">
-    <div className={styles.baseLayout}>{children}</div>
-  </NotificationProvider>
-);
+const getSidebar = (slug: string): FC | false => {
```

Comment:
This unfortunately sounds hacky and non-optimized as hell.

Genuinely speaking this should NOT be on the Base layout. These should be part of the children layouts. I can't approve of this change.

The Base layout should be the simplest piece of layout, and I'm sure there must be alternatives for achieving what you want, without polluting the base layout this way.

Reply:
I agree. I too felt that the Base layout became bloated. I preferred to modify WithLayout and move it into `layout.tsx`, but I can't use Matter context anymore to find layout components. Please let me know if I'm wrong.

Reply:
We could also move all the context stuff from `page.tsx` to `layout.tsx`, use WithLayout in it, and retain the existing children layouts but that would make the layout bloated and potentially cause unexpected issues. Let me know if you have an alternative in mind.